src/providers/mbed_provider/key_management.rs

```diff
@@ -46,42 +45,41 @@ fn create_key_id(
     key_triple: KeyTriple,
     key_attributes: Attributes,
     store_handle: &mut dyn ManageKeyInfo,
-    local_ids_handle: &mut LocalIdStore,
+    max_current_id: &AtomicU32,
 ) -> Result<key::psa_key_id_t> {
-    let mut rng = SmallRng::from_entropy();
-    let mut key_id = rng.gen_range(key::PSA_KEY_ID_USER_MIN, key::PSA_KEY_ID_USER_MAX + 1);
-
-    while local_ids_handle.contains(&key_id) {
-        key_id = rng.gen_range(key::PSA_KEY_ID_USER_MIN, key::PSA_KEY_ID_USER_MAX + 1);
+    // fetch_add adds 1 to the old value and returns the old value, so add 1 for new ID
+    let new_key_id = max_current_id.fetch_add(1, Relaxed) + 1;
```

Comment:
I guess it is fine to not care about overflows as `PSA_KEY_ID_USER_MAX` is equal to the defined value of `0x3FFFFFFF`.

Reply:
Better safe than sorry or overkill? 😉